keras_rs/src/utils/keras_utils.py

```diff
@@ -1,10 +1,32 @@
-from typing import Union
+from typing import Any, Callable, Union
 
 import keras
 
 from keras_rs.src import types
 
 
+def no_automatic_dependency_tracking(
+    fn: Callable[..., Any],
+) -> Callable[..., Any]:
+    """Decorator to disable automatic dependency tracking in Keras and TF.
+
+    Args:
+        fn: the function to disable automatic dependency tracking for.
+
+    Returns:
+        a wrapped version of `fn`.
+    """
+    if keras.backend.backend() == "tensorflow":
+        import tensorflow as tf
+
+        fn = tf.__internal__.tracking.no_automatic_dependency_tracking(fn)
```

Comment:
Quick question, mainly for my understanding: why do we not disable this in core Keras itself?

Reply:
That is a good question. Apparently, we didn't need it.

Normally, this is about not tracking `Layers`, `Metrics` and `Variables`. The reason I need it is not because of that, it's because dicts, lists, etc. get wrapped into `TrackedDict`, `TrackedList` etc. and the Tensorflow equivalents. I need configuration structures that are nested, but then, when I pass the nested configs to some APIs, that API complains about `TrackedDict`, `TrackedList`, etc. All I want is to preserve the original structures.